vwunfinished.py

```diff
@@ -0,0 +1,126 @@
+
+input_parser = parser.add_mutually_exclusive_group(required=True)
+input_parser.add_argument("--path", help="Path to a vimwiki file")
+input_parser.add_argument("--date", help="Use diary file for given date (in YYYY-MM-DD format)")
+input_parser.add_argument("--today", help="Use diary file for today", dest="date",
+                          action="store_const", const=str(datetime.now().date()))
+
+
+def vimwiki_unfinished_tasks(path=None, date=None, section=None, bullets=None, count_sublists=None,
+                             wiki_path=None, diary_dir=None, filetype=None):
+    """
+    Return a count of unfinished tasks in specified vimwiki file
+    Consider this function to be an API. It's attributes and return type will remain backwards compatible.
+
+    :param str path: Path to a vimwiki file
+    :param str date: Use diary file for given date (in YYYY-MM-DD format)
+    :param str section: Count tasks only in specified section (e.g. '== Daily checklist ==')
+    :param list bullets: Bullet symbols, e.g. ["*", "-"]
+    :param bool count_sublists: Do you wish to count sublists?
+    :param str wiki_path: Use nonstandard wiki path (can contain ~ for home directory)
+    :param str diary_dir: Use nonstandard diary directory
+    :param str filetype: Either "wiki" or "md"
+
+    :raise: ValueError if there is not enough information to determine a wiki file
+    :raise: IOError if wiki file doesn't exist
+
+    :return: int
+    """
+    provider = VimwikiFileProvider(path=path, date=date, wiki_path=wiki_path,
+                                   diary_dir=diary_dir, filetype=filetype)
+    counter = UnfinishedTasksCounter(text=provider.content, section=section, bullets=bullets,
+                                     count_sublists=count_sublists)
+    return counter.count_unfinished_tasks()
+
+
+class UnfinishedTasksCounter(object):
+    def __init__(self, text=None, section=None, bullets=None, count_sublists=True):
+        self.bullets = bullets or ["-", "*"]
+        self.section = section
+        self._text = text
+        self.count_sublists = count_sublists
+
+    @property
+    def text(self):
+        if not self.section:
+            return self._text
+        level = self.section.split()[0]
+        start = self._text.find(self.section)
+        end = self._text.find(level, start + len(self.section))
+        return self._text[start:end]
+
+    @property
+    def unfinished_bullet_str(self):
+        return tuple(["{} [ ]".format(x) for x in self.bullets])
+
+    @property
+    def unfinished_tasks(self):
+        tasks = []
+        for line in self.text.split("\n"):
+            if self.count_sublists:
+                line = line.strip()
+            if line.startswith(self.unfinished_bullet_str):
+                tasks.append(line)
+        return tasks
+
+    def count_unfinished_tasks(self):
+        return len(self.unfinished_tasks)
```

Comment:
Another suggestion for future work:  Since you have these anyway, it might be nice to provide an option to print out the full task list.